MODEL-GUIDE.md

```diff
@@ -302,6 +303,79 @@ The build script will enforce the following keyword conventions:
         value: 0.0.0.0
     ```
 
+* x-field-uid
+
+  * Field Unique Identifiers (UIDs) are required for every property/enum/included object.  
+  * They need to be unique (within the object), and once assigned can never be changed/re-used, even if the field itself is deprecated and eventually removed.  
+  * They are required for maintaining protobuff backwards compatibility.  
+  * Should follow protobuff [Assigning Field Numbers](https://developers.google.com/protocol-buffers/docs/proto3#assigning_field_numbers) conventions.
+
+  * Use-Case : Adding a New Property to an Existing Object
+    * Adding a new property 'peer_name' to an existing object.
+    * `x-field-uid` must be added with different value (e.g. 2)
+
+    <img src="images/use_case_1.PNG" alt="drawing" width="500"/>
+
+  * Use-Case : Adding an enum value
+    * enum must be replaced with `x-enum` 
+    * Adding a new enum `fiber` to the existing property field (`media`)
+    * `x-field-uid` must be added with different value (e.g. 2)
+    * The bundler will populate `enum` at the time of bundling. So, it will follow OpenAPI standard. 
+    * `x-enum` remain within final yaml file to generate unique proto uid.
+
+    <img src="images/use_case_2.PNG" alt="drawing" width="500"/>
+
+  * Use-Case : Including a Pre-Existing Object
+    * `x-include` to include only the field of the property (`..#/components/schemas/GlobalObject/properties/name`)
+    * Same property name (`uniquename`) must be used for both base and derived property. 
+    * The bundler will correctly merge the `x-include` (e.g. `type` and `pattern`) and drop the `x-include` keyword from the merged object.
+    * At the time of merge it will not include base `x-field-uid` `x-field-uid: 2` should present in the derived property.
+    
+    <img src="images/use_case_3.PNG" alt="drawing" width="500"/>
+
+  * Use-Case : Deprecating a Property 
+    * Model should maintain `x-field-uid: 1` when property will deprecate 
+    * `x-status` must be added with status deprecated
+
+    <img src="images/use_case_4.PNG" alt="drawing" width="500"/>
+
+  * Use-Case : Removing a Property
+    * User must never use the numbers added in `x-reserved-field-uids` again.
+    * That `x-field-uid` number should include within existing or new `x-reserved-field-uids`
+    * Try to remove `ieee_802_1qbb` which have `x-field-uid: 1`. That number must be included within `x-reserved-field-uids: [1]`
+
+    <img src="images/use_case_5.PNG" alt="drawing" width="500"/>
+
+  * Use-Case : Removing an enum value
+    * User must never use the numbers added in `x-reserved-field-uids` again.
+    * That `x-field-uid` number should include within existing or new `x-reserved-field-uids`
+    * Removing property `b` which have `x-field-uid: 2`. That number must be included within `x-reserved-field-uids: [2]`
+  
+    <img src="images/use_case_6.PNG" alt="drawing" width="500"/>
```

Comment:
Can we update this example so that there is already one x-reserved-field-uid on the left side and after removing the enum we have two x-reserved-field-uid on the right side?  
That way, we provide an example of having a list of x-reserved-field-uid.